atomicfu/src/commonTest/kotlin/kotlinx/atomicfu/test/UncheckedCastTest.kt

```diff
@@ -0,0 +1,28 @@
+package kotlinx.atomicfu.test
+
+import kotlinx.atomicfu.AtomicRef
+import kotlinx.atomicfu.atomic
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+private val topLevelS = atomic<Any>(listOf("A", "B"))
+
+class UncheckedCastTest {
+    private val s = atomic<Any>("AAA")
+    private val bs = atomic<Any?>(null)
+
+    @Test
+    fun testAtomicValUncheckedCast() {
```

Comment:
General test style: `assertEquals` first parameters is "expected value" and the second is "actual", so checks here should be written like this:
```
assertEquals( "AAA", (s as AtomicRef<String>).value,)
```
Let's write them correctly in new tests, will fix old ones later.